### PR TITLE
feat: add new separate GitHub Enterprise Cloud (`ghec`) OpenAPI specification, plus some new APIs and fixes

### PR DESCRIPTION
**Pull request by octokitbot**
This adds new, separate OpenAPI bundles for GitHub Enterprise Cloud. These files will include all API operations available for GHEC customers on `api.github.com`, including GHEC-exclusive API operations and operations available to Free, Pro and Teams customers.

BREAKING CHANGE: Any API operations only available to GitHub Enterprise Cloud (GHEC) customers will no longer appear in the `api.github.com` OpenAPI specifications. These will only appear in the new GHEC-specific (`ghec`) OpenAPI specifications.

A number of other changes are also present:

* feat: add new "Get a Dependabot alert" API (`GET /repos/{owner}/{repo}/dependabot/alerts/{alert_number}`)
* feat: add new "Update a Dependabot alert" API (`PATCH /repos/{owner}/{repo}/dependabot/alerts/{alert_number}`)
* feat: add new "List Dependabot alerts for a repository" API (`GET /repos/{owner}/{repo}/dependabot/alerts`)
* fix: document `404 Not Found` status returned by labels-related APIs
* fix: mark recently-added "List CodeQL databases for a repository" API (`GET /repos/{owner}/{repo}/code-scanning/codeql/databases`) and Get a CodeQL database for a repository" API (`GET /repos/{owner}/{repo}/code-scanning/codeql/databases/{language}`) as available for non-GitHub Enterprise Cloud customers
* fix: document that APIs can return `422` errors when an endpoint has been spammed with many requests
* fix: document `301` status returned by labels-related APIs if the repo has been renamed or moved
* fix: document that "Delete a GitHub Pages site" (`DELETE  /repos/{owner}/{repo}/pages`) and "Update information about a GitHub Pages site" (`PUT /repos/{owner}/{repo}/pages`) APIs return `409` if the repo is archived

### CODE REVIEW

**Comment by timrogers** [comment]
Screenshot 2022-09-19 at 13 37 47

😭

**Comment by gr2m** [comment]
Looks like mostly GHEC changes. It's going to take some work to work through it, but it's great that it's hear. We no longer need the overlay, it will unblock https://github.com/octokit/plugin-enterprise-cloud.js/. It's been a while, but I think there was a lot of complexities due to the fact that there was no dedicated GHEC OpenAPI spec

**Comment by timrogers** [comment]
It does seem like GitHub is being a bit weird as the diff is not _that_ big!

I did start looking at this, but I need to come back to it when I have some more time.

What does this unblock for the EC plugin?

**Comment by gr2m** [comment]
> What does this unblock for the EC plugin?

I _think_ we used to have an overlay for GHEC within the dotcom OpenAPI spec. That overlay was removed, but without introducing a dedicated GHEC OpenAPI spec at the same time. So we've been in limbo for a while now. Once we get this merged and released, we can wire up https://github.com/octokit/plugin-enterprise-cloud.js for auto-updates again

**Review by timrogers — APPROVED** [review]

**Pull request merged**
